zingolib/src/wallet/transaction_context.rs

```diff
@@ -174,6 +175,48 @@ pub mod decrypt_transaction {
             }
         }
 
+        /// INVARIANT: All transaction from the mempool are received.
+        /// If the transaction was created locally and broadcast then:
+        ///  * it is in the database "trbid", unless that structure was wiped
+        ///  * if memory was lost after broadcast, we'll just wait for a transaction
+        ///     that we broadcast to show up on chain, after all until it's validated..
+        ///     it's just gossip, this is an edge case we are safe ignoring
```

Comment:
i believe that if the app is closed, a pending transaction added due to broadcasting would not be stored and when the app is restarted this tx would not appear in the wallet. when the broadcast tx is received again from the mempool it would satisfy the `is_none()` logic and then be re-added to the wallet. therefore, this comment is maybe misleading us to believe that the pending tx will not re-appear if the memory was lost

Reply:
i have just realised that you have removed some scanning logic such as adding outgoing_tx_datas from the mempool scanning. this will appear very buggy when users close and re-open the app etc. as sending transactions will change to receiving etc.

i understand that your assumption is that mempool is only for receiving transactions, but we also use the mempool to restore pending transactions that we have broadcast. I don't see any reason why calling scan_full_tx is a problem here but calling this new function will cause bugs on reload

Reply:
Per your point that my assumptions about `rtransactions` from the mempool not always being received, and in the case that they're forgotten-sent, the app misbehaving.

I'm not surprised, and agree that we need to improve the code to properly handle the case.   I wasn't totally clear on correct behavior, and wrote that doc-comment explicitly calling out the edge case in the full expectation that it might well be innacurate.